src/modules/task-show.js

```diff
@@ -0,0 +1,130 @@
+
+  toDoTasks.forEach((key) => {
+    /* || Get all values */
+    const toDoValue = localStorage.getItem(key);
+
+    const toDoInnerDiv = document.createElement('p');
+    toDoInnerDiv.classList.add('p__cont__task');
+    toDoInnerDiv.setAttribute('id', key);
+    toDoInnerDiv.innerHTML = `<p id="todo__name">${key.split('_')[1]}</p>`;
+    /* || localstorage would store boolean as string so we parse it to boolean back */
+    const buttonModify = document.createElement('button');
+    buttonModify.classList.add('edit__btn');
+    buttonModify.innerHTML = '<i class="fa-solid fa-pen-to-square"></i>';
+    if (!JSON.parse(toDoValue)) {
+      buttonModify.style.visibility = 'visible';
+    } else {
+      buttonModify.style.visibility = 'hidden';
+      toDoInnerDiv.classList.add('completedTask');
+    }
+    toDoInnerDiv.appendChild(buttonModify);
+    toDoInnerDiv.innerHTML += '<button class="del__btn"><i class="fa-solid fa-trash"></i></button>';
+    toDoDivTask.appendChild(toDoInnerDiv);
+  });
+  /* || tasks completed */
+  toDoTasks = document.querySelectorAll('.p__cont__task');
+  toDoTasks.forEach((element) => {
+    element.onclick = () => {
+      /*  || local storage update */
+      if (element.classList.contains('completedTask')) {
+        storageUpdate(element.id.split('_')[0], element.innerText, false);
+      } else {
+        storageUpdate(element.id.split('_')[0], element.innerText, true);
+      }
+    };
+  });
+  /* || Edit Tasks */
+  modifyToDoTasks = document.getElementsByClassName('edit__btn');
+  Array.from(modifyToDoTasks).forEach((element) => {
+    element.addEventListener('click', (e) => {
+      e.stopPropagation();
+      /* || disable other edit buttons when one task is being edited */
+      buttonsDeactive(true);
+      /* || update input value and remove div */
+      const parentEl = element.parentElement;
+      inputNewTasks.value = parentEl.querySelector('#todo__name').innerText;
+      /* || set updateNote to the task that is being edited */
+      toDotasksupdate = parentEl.id;
+      /*  || remove task */
+      parentEl.remove();
+    });
+  });
+  /*   || Delete Tasks */
+  clearToDoTasks = document.getElementsByClassName('del__btn');
+  Array.from(clearToDoTasks).forEach((element) => {
+    element.addEventListener('click', (e) => {
+      e.stopPropagation();
+      /* || Delete from local storage and remove div */
+      const parentEl = element.parentElement;
+      deleteTask(parentEl.id);
+      parentEl.remove();
+      counter -= 1;
+    });
+  });
+};
+/* || clear all checked tasks */
+deletToDoTasks.addEventListener('click', (e) => {
+  counter = Object.keys(localStorage).length;
+  e.stopPropagation();
+  const tasks1 = document.querySelectorAll('.p__cont__task');
+  tasks1.forEach((element) => {
+    /* || local storage update */
+    if (element.classList.contains('completedTask')) {
+      deleteTask(element.id);
+      element.remove();
+      counter -= 1;
+    }
+  });
```

Comment:
- [ ] Clear all completed tasks functionality should use the Array.filter() function. This is a higher-order function that takes a function and called on an array then returns a new array. Please update your logic in order to make use of it. As I said, your tasks are not saved in an array therefore it will require you to first put in array and then the returned array is the one that is again saved in the local storage as an update 😉